extendr-api/src/graphics/device_driver.rs

```diff
@@ -0,0 +1,827 @@
+                confirm
+            } else {
+                false.into()
+            }
+        }
+
+        unsafe extern "C" fn device_driver_holdflush<T: DeviceDriver>(
+            dd: pDevDesc,
+            level: c_int,
+        ) -> c_int {
+            let data = ((*dd).deviceSpecific as *mut T).as_mut().unwrap();
+            data.holdflush(*dd, level as _)
+        }
+
+        #[cfg(use_r_ge_version_14)]
+        unsafe extern "C" fn device_driver_setPattern<T: DeviceDriver>(
+            pattern: SEXP,
+            dd: pDevDesc,
+        ) -> SEXP {
+            let data = ((*dd).deviceSpecific as *mut T).as_mut().unwrap();
+            // TODO
+            // data.setPattern(pattern, *dd)
+            R_NilValue
+        }
+
+        #[cfg(use_r_ge_version_14)]
+        unsafe extern "C" fn device_driver_releasePattern<T: DeviceDriver>(
+            ref_: SEXP,
+            dd: pDevDesc,
+        ) {
+            let data = ((*dd).deviceSpecific as *mut T).as_mut().unwrap();
+            // TODO
+            // data.reelasePattern(ref_, *dd);
+        }
+
+        #[cfg(use_r_ge_version_14)]
+        unsafe extern "C" fn device_driver_setClipPath<T: DeviceDriver>(
+            path: SEXP,
+            ref_: SEXP,
+            dd: pDevDesc,
+        ) -> SEXP {
+            let data = ((*dd).deviceSpecific as *mut T).as_mut().unwrap();
+            // TODO
+            // data.setClipPath(path, ref_, *dd)
+            R_NilValue
+        }
+
+        #[cfg(use_r_ge_version_14)]
+        unsafe extern "C" fn device_driver_releaseClipPath<T: DeviceDriver>(
+            ref_: SEXP,
+            dd: pDevDesc,
+        ) {
+            let data = ((*dd).deviceSpecific as *mut T).as_mut().unwrap();
+            // TODO
+            // data.releaseClipPath(ref_, *dd);
+        }
+
+        #[cfg(use_r_ge_version_14)]
+        unsafe extern "C" fn device_driver_setMask<T: DeviceDriver>(
+            path: SEXP,
+            ref_: SEXP,
+            dd: pDevDesc,
+        ) -> SEXP {
+            let data = ((*dd).deviceSpecific as *mut T).as_mut().unwrap();
+            // TODO
+            // data.setMask(path, ref_, *dd)
+            R_NilValue
+        }
+
+        #[cfg(use_r_ge_version_14)]
+        unsafe extern "C" fn device_driver_releaseMask<T: DeviceDriver>(ref_: SEXP, dd: pDevDesc) {
+            let data = ((*dd).deviceSpecific as *mut T).as_mut().unwrap();
+            // TODO
+            // data.releaseMask(ref_, *dd);
+        }
+
+        //
+        // ************* defining the wrapper functions ends here ****************
+        //
+
+        // `Box::new()` allocates memory on the heap and place `self` into it.
+        // Then, an unsafe function `Box::into_raw()` converts it to a raw
+        // pointer. By doing this, Rust won't drop the object so that it will
+        // live after passing it to the R's side. In other words, extendr should
+        // take care of dropping in the `close()` wrapper.
```

Comment:
I suggest 'By doing *so*' and 'it will *survive* after', and perhaps 'after *being passed* to the R's side'.